### ISSUE TITLE
[DOCS] Fails to add predictions to dataset using torchvision's model

### ISSUE THREAD

**Issue by stevezkw1998** [voxel51/fiftyone#2825]
### URL(s) with the issue

[Please provide a link to the documentation entry in question.](https://docs.voxel51.com/tutorials/evaluate_detections.html#Add-predictions-to-dataset)

### Description of proposal (what needs changing)

I try the code example but doesn't have the same effect.

script
```python
import fiftyone as fo
# You must import fiftyone as fo first, otherwise there will be
# OMP: Error #15: Initializing libiomp5md.dll, but found libiomp5md.dll already initialized.
# not sure why
import torch
import torchvision


from PIL import Image
import torchvision.transforms.functional as func

dataset = fo.zoo.load_zoo_dataset(
    "coco-2017",
    split="validation",
    dataset_name="evaluate-detections-tutorial",
    label_types=["detections", "segmentations"],
    classes=["cat", "dog"],
    max_samples=50,
    shuffle=False,
)
dataset.persistent = True

# Run the model on GPU if it is available
device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")

model = torchvision.models.get_model('fasterrcnn_resnet50_fpn')
model.to(device)
model.eval()

classes = dataset.default_classes

# Add predictions to samples
with fo.ProgressBar() as pb:
    for sample in pb(dataset):
        # Load image
        image = Image.open(sample.filepath)
        image = func.to_tensor(image).to(device)
        c, h, w = image.shape

        # Perform inference
        preds = model([image])[0]
        labels = preds["labels"].cpu().detach().numpy()
        scores = preds["scores"].cpu().detach().numpy()
        boxes = preds["boxes"].cpu().detach().numpy()

        # Convert detections to FiftyOne format
        detections = []
        for label, score, box in zip(labels, scores, boxes):
            # Convert to [top-left-x, top-left-y, width, height]
            # in relative coordinates in [0, 1] x [0, 1]
            x1, y1, x2, y2 = box
            rel_box = [x1 / w, y1 / h, (x2 - x1) / w, (y2 - y1) / h]

            detections.append(
                fo.Detection(
                    label=classes[label],
                    bounding_box=rel_box,
                    confidence=score
                )
            )

        # Save predictions to dataset
        sample["predictions"] = fo.Detections(detections=detections)
        sample.save()

session = fo.launch_app(dataset)
```
log
```
Downloading split 'validation' to 'C:\Users\a4781\fiftyone\coco-2017\validation' if necessary
Found annotations at 'C:\Users\a4781\fiftyone\coco-2017\raw\instances_val2017.json'
Sufficient images already downloaded
Existing download of split 'validation' is sufficient
Loading 'coco-2017' split 'validation'
 100% |██████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [504.9ms elapsed, 0s remaining, 99.1 samples/s]      
Dataset 'evaluate-detections-tutorial' created
 100% |██████████████████████████████████████████████████████████████████████████████████████████████████████| 50/50 [1.7m elapsed, 0s remaining, 0.5 samples/s]     
App launched. Point your web browser to http://localhost:5151
```
My results
![image](https://user-images.githubusercontent.com/53121066/226746389-0470bb19-2eae-4ebd-8af8-8ca0ef2e2b0c.png)
You can see the labels 'predictions' has 0 objects.

You can see this fails in `preds = model([image])[0]`
```python
In [23]: image
Out[23]:
tensor([[[0.6667, 0.6784, 0.6863,  ..., 0.2706, 0.2667, 0.2745],
         [0.6745, 0.6902, 0.6941,  ..., 0.2706, 0.2824, 0.2784],
         [0.6863, 0.6941, 0.6980,  ..., 0.2745, 0.2706, 0.2784],
         ...,
         [0.7373, 0.7176, 0.7569,  ..., 0.7294, 0.7294, 0.7333],
         [0.7294, 0.7333, 0.7294,  ..., 0.7765, 0.7647, 0.7294],
         [0.7294, 0.7333, 0.7294,  ..., 0.5059, 0.4941, 0.4196]],

        [[0.5333, 0.5569, 0.5647,  ..., 0.2980, 0.2980, 0.2784],
         [0.5529, 0.5686, 0.5725,  ..., 0.3020, 0.3137, 0.2941],
         [0.5647, 0.5725, 0.5765,  ..., 0.3059, 0.3020, 0.2941],
         ...,
         [0.7412, 0.7176, 0.7333,  ..., 0.6157, 0.6157, 0.6118],
         [0.7176, 0.7216, 0.7176,  ..., 0.5255, 0.4706, 0.3451],
         [0.7176, 0.7216, 0.7176,  ..., 0.2353, 0.2235, 0.1608]],

        [[0.2863, 0.3020, 0.3098,  ..., 0.1647, 0.1529, 0.1451],
         [0.3020, 0.3137, 0.3176,  ..., 0.1569, 0.1686, 0.1569],
         [0.3098, 0.3176, 0.3137,  ..., 0.1529, 0.1569, 0.1569],
         ...,
         [0.6157, 0.5843, 0.6000,  ..., 0.6000, 0.6000, 0.6039],
         [0.5961, 0.6000, 0.5961,  ..., 0.5255, 0.4863, 0.3961],
         [0.5882, 0.5922, 0.5961,  ..., 0.2471, 0.2353, 0.1765]]])

In [24]: model([image])
Out[24]: 
[{'boxes': tensor([], size=(0, 4), grad_fn=<StackBackward0>),
  'labels': tensor([], dtype=torch.int64),
  'scores': tensor([], grad_fn=<IndexBackward0>)}]
```

### Willingness to contribute

The FiftyOne Community encourages documentation contributions. Would you or another member of your organization be willing to contribute a fix for this documentation issue to the FiftyOne codebase?

-   [ ] Yes. I can contribute a documentation fix independently
-   [x] Yes. I would be willing to contribute a documentation fix with guidance from the FiftyOne community
-   [ ] No. I cannot contribute a documentation fix at this time

**Comment by stevezkw1998** [comment]
Follow-up:
I found out the reason if I use
model = torchvision.models.get_model('fasterrcnn_resnet50_fpn')
Then I will fail, but if I use
model = torchvision.models.detection.fasterrcnn_resnet50_fpn(pretrained=True) 
Then I will succeed, I am curious about the reason why that two commands will have different effects?
